### ISSUE TITLE
Changelings who died in round retain their abilities when taking bitrunner roles

### ISSUE THREAD

**Issue by DaydreamIQ** [tgstation/tgstation#84765]
Reporting client version: 515.1641

## Round ID:
231770

## Testmerges:
- [[TM ONLY] Hijacked Revision Date for v3 TGS Api](https://www.github.com/tgstation/tgstation/pull/83227)
- [Offset render relays for non-offsetting planes to match highest matching render plane](https://www.github.com/tgstation/tgstation/pull/84184)
- [tgui: custom font bluescreen fix, typescript'd](https://www.github.com/tgstation/tgstation/pull/84601)

## Reproduction:
![image](https://github.com/tgstation/tgstation/assets/62606051/bb8486fb-82d7-49e1-ba17-f99f0cfc8fa9)
A changeling that died in this round went islander and was still able to use all their changeling abilities despite it being a ghostrole

**Comment by BowlOfNoodlez** [comment]
Yep that was me, i had gotten incinerated due to botched friendly ling activities and chose this ghost role, to my surprise still had ling powers, was funny for the short time i had it

**Comment by Sadboysuss** [comment]
this is likely because changeling antag datum is defined on the soul
